### DISCUSSION TITLE
Why is the AssemblyVersion of Microsoft.Bcl.TimeProvider version 8.0.0.1 and not 8.0.1?

### DISCUSSION THREAD

**Discussion by MagicAndre1981** [dotnet/runtime#104712] [Q&A] [answered]
In PR #94459 the [ServicingVersion was set to 1](https://github.com/dotnet/runtime/pull/94459/files#diff-c0aed6def31bb6eadc05d056b1c52f13d3655b81c2e52ef6b7dd2d525e9a98c5R18) but compared to other packages where ServicingVersion number is the 3rd part of the version, here it is the 4th part of the version:

![image](https://github.com/dotnet/runtime/assets/8274816/9bb8b163-2ba4-4057-a9a7-3fd28f140b9a)

But the ProducVersion is 8.0.1 + hash (8.0.1+bf5e279d9239bfef5bb1b8d6212f1b971c434606)

![image](https://github.com/dotnet/runtime/assets/8274816/ead67f7b-5597-4ab0-9a12-e0ba15cde866)

Is this by design?

**Comment by huoyaoyuan** [comment, marked as answer]
The file version, assembly version and NuGet package version are completely independent numbers. They follow different contracts to satisfy different requirements.

> but compared to other packages where ServicingVersion number is the 3rd part of the version

Those are NuGet package versions, which follows SemVer.

/cc @dotnet/area-infrastructure-libraries

**Reply by MagicAndre1981** [reply]
thanks for that information. Why don't all dotnet libs follow SemVer version schema? @ViktorHofer (you marked it as answer, can you explain this?)

**Reply by huoyaoyuan** [reply]
> Why don't all dotnet libs follow SemVer version schema?

There's no difference between particular dotnet libs.

Assembly version is 4 int16 digits and respected by CLR assembly loader. It's incompatible with SemVer which use strings, and originates much earlier.
